### DISCUSSION TITLE
群晖docker部署成功，但节点无法订阅，请大佬指导

### DISCUSSION THREAD

**Discussion by xy2yp** [vvbbnn00/WARP-Clash-API#129] [Q&A]
大佬，我在群晖docker里面拉取镜像并且成功启动了，访问IP:21001也成功。
然后我通过群晖的域名进行反代，经过域名访问这个地址，网页也可以打开。反代地址大概是https://我的域名:反代端口。
这时候页面的订阅地址，也自动从我的IP:21001/api/clash......变成了域名:反代端口/api/clash......
把这个地址放进安卓手机的clash，订阅的失败，提示是
Get "https://反代域名:反代端口/api/clash ?best=true&randomName=true&proxyFormat =full&ipv6=false&key=密码": io: read/write on closed pipe

不知道是哪里错了？请指导下，感谢！

**Comment by xy2yp** [comment]
另外，我看日志，有写着
Execution of job "doAddDataTaskOnce (trigger: interval[0:01:00], next run at: 202
4-03-04 17:38:27 CST)" skipped: maximum number of running instances reached (1)
这个是哪里有问题吗？

**Reply by vvbbnn00** [reply]
这个是上一个任务没有执行完，同时只能执行一个任务

**Comment by vvbbnn00** [comment]
> 大佬，我在群晖docker里面拉取镜像并且成功启动了，访问IP:21001也成功。 然后我通过群晖的域名进行反代，经过域名访问这个地址，网页也可以打开。反代地址大概是[https://我的域名:反代端口。](https://%E6%88%91%E7%9A%84%E5%9F%9F%E5%90%8D:%E5%8F%8D%E4%BB%A3%E7%AB%AF%E5%8F%A3%E3%80%82) 这时候页面的订阅地址，也自动从我的IP:21001/api/clash......变成了域名:反代端口/api/clash...... 把这个地址放进安卓手机的clash，订阅的失败，提示是 Get "[https://反代域名:反代端口/api/clash](https://%E5%8F%8D%E4%BB%A3%E5%9F%9F%E5%90%8D:%E5%8F%8D%E4%BB%A3%E7%AB%AF%E5%8F%A3/api/clash) ?best=true&randomName=true&proxyFormat =full&ipv6=false&key=密码": io: read/write on closed pipe
> 
> 不知道是哪里错了？请指导下，感谢！

在安卓手机的浏览器访问一下，看看能不能访问，是会报错还是直接连不上。如果是直接连不上的话，说明反代可能存在一些问题。

**Reply by xy2yp** [reply]
大佬，试了下，反代页面在手机上访问正常

**Reply by vvbbnn00** [reply]
emmm 那不应该呀 是不是开着代理更新配置文件了？

**Reply by xy2yp** [reply]
没有😂😂😂
我在内网环境试了一下，也是获取不到配置，我在想有没有可能是手机版的clash不支持。
再用软路由上的试了下，能够获取到节点，说明反代应该是没问题。。但是测速全部超时。。

**Reply by vvbbnn00** [reply]
使用的clash是哪个版本，是clash meta吗

**Reply by xy2yp** [reply]
open clash 有启用Meta 内核

**Reply by xy2yp** [reply]
大佬，手机换了一个clash版本之后可以了。
另外还有个问题请教一下，我有一个自己的24PB key，需要怎么导入？群晖docker部署的

**Reply by vvbbnn00** [reply]
可以参考：https://github.com/vvbbnn00/WARP-Clash-API?tab=readme-ov-file#-%E8%BF%9B%E9%98%B6%E6%93%8D%E4%BD%9C
来配置`licenseKey`

**Reply by xy2yp** [reply]
emm...我看到这个curl -X POST http://host:port/api/account/update_license -H "Content-Type: application/json" -d "{\"license_key\": \"your_license_key\"}"

但是不知道应该在哪里输入这个命令😂😂😂可以通过环境变量的形式来配置吗？

**Reply by xy2yp** [reply]
emm...我看到这个curl -X POST http://host:port/api/account/update_license -H "Content-Type: application/json" -d "{\"license_key\": \"your_license_key\"}"

但是不知道应该在哪里输入这个命令😂😂😂可以通过环境变量的形式来配置吗？

**Reply by vvbbnn00** [reply]
这个是发送http请求的意思，你可以找一个发送请求的工具（例如Postman）发送，也可以直接在linux/mac的终端输入这行指令

**Reply by xy2yp** [reply]
好的，我试试，谢谢大佬

**Reply by xy2yp** [reply]
大佬，又有新的问题了。。
我在环境变量里面配置了DO_GET_WARP_DATA为False，但是看日志里面，写的还是DO_GET_WARP_DATA is true。
![Screenshot_20240305_164506](https://github.com/vvbbnn00/WARP-Clash-API/assets/161705149/795c149a-71a7-47b4-8507-28633c96e0ae)

![Screenshot_20240305_164525](https://github.com/vvbbnn00/WARP-Clash-API/assets/161705149/64aa96b5-2ad9-46d4-b155-1942b121ed5e)

**Reply by xy2yp** [reply]
直接在ssh是不是不可以？提示403。

**Reply by vvbbnn00** [reply]
如果设置了SECRET_KEY，需要在url后面加上`?key=xxx`

**Reply by vvbbnn00** [reply]
这个docker问题有点奇怪，理论上应该已经关闭了才对（
你试试在docker内部，输入env，检查一下环境变量，或者重启docker容器试试

**Reply by xy2yp** [reply]
大佬，我先在环境变量里面把secret删掉了，然后POST了命令，成功修改了licensekey。然后再重新把secret配置上。
通过  群晖ip:21001打开页面，通过里面的链接，http://IP:21001/api/clash……key=密码，可以获取到clash配置，也正常上网。延迟大概在300ms上下， 主打一个优秀。
但是如果我通过反代的网页，域名:反代端口，通过里面的链接，https://域名:反代端口，获取配置失败，提示 profile does not contain 'proxies' or proxy-providers'。通过网页上下载yaml文件来导入，可以导入成功，但是8个节点只有2个能通，延迟都在800ms以上。。搞不懂。。。

**Reply by xy2yp** [reply]
破案了！我的反代网址开启了Auth认证，关掉就能获取到了。
（果然浴室深思是有用的

**Reply by xy2yp** [reply]
关于DO_GET_WARP_DATA的问题，没想到办法，于是把时间间隔设成86400S
